src/main/treeDataProviders/dependenciesTree/dependenciesRoot/goTree.ts

```diff
@@ -40,52 +37,88 @@ export class GoTreeNode extends RootNode {
         }
         this.generalInfo = new GeneralInfo(rootPackageName, '', ['None'], this.workspaceFolder, GoUtils.PKG_TYPE);
         this.label = rootPackageName;
-        if (goList.length === 0) {
+        if (goModGraph.length === 0) {
             return;
         }
-        this.buildDependenciesMapAndDirectDeps(goList);
-        this.children.forEach(child => this.populateDependenciesTree(child, quickScan));
+        let dependenciesMap: Map<string, string[]> = this.buildDependenciesMapAndDirectDeps(goModGraph, goList);
+        this.children.forEach(child => this.populateDependenciesTree(dependenciesMap, child, quickScan));
     }
 
-    private buildDependenciesMapAndDirectDeps(goList: string[]) {
-        let i: number = 0;
+    /**
+     * Run "go mod graph" in order to create the dependency tree later on.
+     * @returns a list of dependencies in the following order:
+     * For a given index i, if i is even (i%2==0) results[i] is the package that depends on results[i+1]: results[i] -> results[i+1].
+     * The first lines of the even indices contain no versions. Those are the direct dependencies.
+     */
+    private runGoModGraph(): string[] {
```

Comment:
Instead of using a less-trial output structure for this function, I recommend we use a class with two fields, with self-explanatory names, instead of the even / uneven array structure.